### ISSUE TITLE
Implement a new polarity classifier

### ISSUE THREAD

**Issue by mark-torres10** [METResearchGroup/social_agent_simulation_platform#213]
# Issue description

Basically the same as https://github.com/METResearchGroup/social_agent_simulation_platform/issues/212 and https://github.com/METResearchGroup/social_agent_simulation_platform/issues/205, but for implementing a polarity ML classifier.

Other related resources to check out:
- https://github.com/METResearchGroup/social_agent_simulation_platform/blob/main/strategy_planning/2026-03-13_feature_extraction_api/FEATURE_EXTRACTION_API_PRD.md
- https://github.com/METResearchGroup/social_agent_simulation_platform/blob/main/strategy_planning/2026-03-13_feature_extraction_api/INTERN_PROJECT_CONTRACT.md

## Description

We want to track the polarity of posts. This is split into three labels, "positive", "negative", and" neutral".

## Model

Let's use `cardiffnlp/twitter-roberta-base-sentiment-latest `, a RoBERTa-base sentiment classifier trained on ~124M English tweets (2018–2021) and finetuned on TweetEval sentiment (3-way: negative/neutral/positive).

## Current requirements

Add a small module under `ml_tooling/` that provides a stable Python API.

The code should look like this:

```python
ml_tooling/polarity/
   models.py # put pydantic models here
   classifier.py # put classifier here
   examples_test.py # put examples here that can be run as a one-off script.
   tests/
      test_classifier.py # put pytest unit tests here. These will be run in CI.
```

### Public API

- `extract_polarity(text: str) -> PolarityLabel`
- `extract_polarity_batch(texts: list[str]) -> list[PolarityLabel]`
- Can implement batch like you did for https://github.com/METResearchGroup/social_agent_simulation_platform/issues/205.

### Output type

Use a Pydantic model with:

```python
class PolarityValue(enum): # or something like this, I forget the exact syntax, but create an enum to define the PolarityValue as positive/negative/neutral
   

class PolarityLabel:
   text_id: str
   text: str
   polarity_label: PolarityValue
   polarity_prob: float # probability of the highest label class
   label_timestamp: str # use lib/timestamp_utils.py to get the timestamp
```
### Basic implementation example

This is just a code snippet for how to use the model, to show how it works:

```python
from transformers import AutoTokenizer, AutoModelForSequenceClassification
import torch
from scipy.special import softmax

MODEL_NAME = "cardiffnlp/twitter-roberta-base-sentiment-latest"

tokenizer = AutoTokenizer.from_pretrained(MODEL_NAME)
model = AutoModelForSequenceClassification.from_pretrained(MODEL_NAME)

# Optional: simple tweet-style preprocessing
def preprocess(text: str) -> str:
    tokens = []
    for t in text.split(" "):
        if t.startswith("@") and len(t) > 1:
            t = "@user"
        if t.startswith("http"):
            t = "http"
        tokens.append(t)
    return " ".join(tokens)

text = "I love using this model on Twitter data!"
text_prep = preprocess(text)

encoded = tokenizer(text_prep, return_tensors="pt")
with torch.no_grad():
    output = model(**encoded)

scores = output.logits[0].numpy()
probs = softmax(scores)

labels = ["negative", "neutral", "positive"]
for label, p in zip(labels, probs):
    print(label, float(p))
```
